final_task/rss_reader/functions/print_func.py

```diff
@@ -0,0 +1,100 @@
+"""Contain formatting output functions"""
+
+import json
+import classes.exceptions as exc
+
+
+def check_limit_argument(command_line_args, news_collection, logger):
+    """Check presence and validness of limit argument
+
+    If present and valid, then limit collection of news
+    """
+    if command_line_args.limit or command_line_args.limit == 0:
+        limit = command_line_args.limit
+        if limit < 0:
+            raise exc.LimitArgumentError("Limit argument can't be negative!")
+        elif len(news_collection) > limit >= 0:
+            return news_collection[:limit]
+        else:
+            logger.warning("Limit argument is longer then list of avaliable news")
+            return news_collection
+    else:
+        return news_collection
+
+
+def generate_news_json(news_collection, logger):
+    """Create json of news"""
+    all_news_dict = {
+                    'Feed': news_collection[0].feed_title,
+                    'news': []
+                    }
+    logger.info("Configure json file...")
+    for news in news_collection:
+        news_dict = {}
+        news_dict['title'] = news.title
+        news_dict['date'] = news.date
+        news_dict['link'] = news.link
+        news_dict['text'] = news.text
```

Comment:
It is better to use method `asdict` from `dataclass` library for this. Check it out

Reply:
Fixed

Reply:
Well, when I was implementing fifth iteration, i was enforced to redefine asdict(), because News class became contain argparse.Namespace class instance. And I didn't need it in the dictionary.

Reply:
It is better to not redefine anything from standart libs, you could just `.pop()` anything you do not need from the dict